### PR TITLE
Dynamic comparison for any two branches added

### PR DESCRIPTION
**Pull request by karan-batavia**

### CODE REVIEW

File: .github/workflows/comparison-results.yml [hunk 1]
```diff
@@ -92,37 +180,85 @@ jobs:
...
         uses: MeilCli/slack-upload-file@v3
         with:
           slack_token: ${{ secrets.SLACK_TOKEN }}
           channel_id: ${{ secrets.SLACK_CHANNEL_ID }}
-          file_path: '/home/runner/work/privado-core/privado-core/comparison_report.csv'
-          initial_comment: 'Comparison Results generated.'
-          file_type: 'csv'
\ No newline at end of file
+          file_path: "/home/runner/work/privado-core/privado-core/report/comparison_report.csv"
+          initial_comment: "Comparison Results generated on ${{github.event.repository.name}} by PR ${{github.event.number}} from branch ${{github.head_ref}} to ${{github.base_ref}}"
```
Comment (ojaswa1942) [inline .github/workflows/comparison-results.yml:263]: Can we have a link here too?

**Comment by ojaswa1942** [comment]
@karan-batavia Overall looks good. Have we run this elsewhere? If so, can you share the workflow logs?

File: .github/workflows/comparison-results.yml (hunk 1 above)
Comment (karan-batavia) [reply .github/workflows/comparison-results.yml]: Done

**Comment by ojaswa1942** [comment]
I see 21 warnings in the workflow run, most related to using the unsupported attribute `owner`.
```
      - name: Clone Privado-Inc/privado
        uses: actions/checkout@v2
        with:
          owner: "Privado-Inc"
          repository: "Privado-Inc/privado"
          path: ./privado-core-and-rules/privado
```

Let's fix these.

**Comment by karan-batavia** [comment]
> I see 21 warnings in the workflow run, most related to using the unsupported attribute `owner`.
> 
> ```
>       - name: Clone Privado-Inc/privado
>         uses: actions/checkout@v2
>         with:
>           owner: "Privado-Inc"
>           repository: "Privado-Inc/privado"
>           path: ./privado-core-and-rules/privado
> ```
> 
> Let's fix these.

All warnings gone.
https://github.com/karan-batavia/privado-core/actions/runs/3729680888/jobs/6325857762

**Review by ojaswa1942 — APPROVED** [review]
:shipit:

**Comment by karan-batavia** [comment]
Thanks, should I merge them?

**Pull request merged**